src/VisualStudio/Core/Def/Implementation/SymbolSearch/SymbolSearchSource.cs

```diff
@@ -0,0 +1,99 @@
+
+            var symbolSearchContext = new SymbolSearchContext(this, callback, rootNodeName, token);
+
+            try
+            {
+                switch (navigationKind)
+                {
+                    case PredefinedNavigationKinds.Definition:
+                        {
+                            // This is not yet supported,
+                            // because Roslyn implementation depends on UI thread.
+                            return SymbolSearchStatus.Withheld;
+                            // Existing Roslyn implementation, disabled to not interfere with demos
+                            /*
+                            var goToDefinitionService = roslynDocument.GetLanguageService<IGoToDefinitionService>();
+                            goToDefinitionService.TryGoToDefinition(roslynDocument, sourceLocation.PersistentSpan.Span.GetStartPoint(snapshot).Position, token);
+                            return SymbolSearchStatus.Completed;
+                            */
+                        }
+                    case PredefinedNavigationKinds.Implementation:
+                        {
+                            var findUsagesService = roslynDocument.GetLanguageService<IFindUsagesService>();
+                            var context = new SimpleFindUsagesContext(token);
+                            await findUsagesService.FindImplementationsAsync(roslynDocument, sourceLocation.PersistentSpan.Span.GetStartPoint(snapshot).Position, symbolSearchContext)
+                                .ConfigureAwait(false);
+                            return SymbolSearchStatus.Completed;
+                        }
+                    case PredefinedNavigationKinds.Reference:
+                        {
+                            var findUsagesService = roslynDocument.GetLanguageService<IFindUsagesService>();
+                            var context = new SimpleFindUsagesContext(token);
+                            await findUsagesService.FindReferencesAsync(roslynDocument, sourceLocation.PersistentSpan.Span.GetStartPoint(snapshot).Position, symbolSearchContext)
+                                .ConfigureAwait(false);
+                            return SymbolSearchStatus.Completed;
+                        }
+                    case PredefinedNavigationKinds.Base:
+                        {
+                            var goToBaseService = roslynDocument.GetLanguageService<IGoToBaseService>();
+                            var context = new SimpleFindUsagesContext(token);
+                            await goToBaseService.FindBasesAsync(roslynDocument, sourceLocation.PersistentSpan.Span.GetStartPoint(snapshot).Position, symbolSearchContext)
+                                .ConfigureAwait(false);
+                            return SymbolSearchStatus.Completed;
+                        }
+                    default:
+                        return SymbolSearchStatus.Withheld;
```

Comment:
what does Withheld mean?

Reply:
That a source could give items, but did not due to high cost, and user needs to make this request explicitly. I changed this to Failed because there's no way around it.